cypress/integration/cucumber/step_definitions/refapp-3.x/01-login/login.js

```diff
@@ -0,0 +1,35 @@
+import {Given} from 'cypress-cucumber-preprocessor/steps';
+
+Given('user arrives at the login page', () => {
+    cy.visitPage('/login');
+})
+
+When('the user logs in with {string} and {string} to the {string}', (username, password, location) => {
+    cy.get('#username').type(username);
+    cy.contains('Continue').click();
+    cy.get('#password').type(password);
+    cy.contains('Log in').click();
+    // The user will be redirected to the location selection page if the credentials are correct.
+    // The following artificial wait is to wait until it changes the url.
+    cy.wait(2000);
```

Comment:
Don't use static `wait`s. Please see [the Cypress documentation](https://docs.cypress.io/faq/questions/using-cypress-faq#How-can-I-make-Cypress-wait-until-something-is-visible-in-the-DOM).